tests/arc/t19795.nim

```diff
@@ -0,0 +1,11 @@
+type Node = ref object
+  kind: Node
+
+var n1 = Node()
+var n2 = Node()
+n2.kind = n1
+
+proc foo =
+  echo cast[int](n1)
```

Comment:
That happens to work now that `n1` is a global but in general for a `cast` expression of type `int` there is no destructor involved. When `n` is a cast expression we need to check for `n.typ` not for `n[0].typ`! Thus maybe the real bug is in `skipConvDfa`.

Reply:
am not sure I understand correct or not, maybe a test case trigger the issue would help.